synapse/storage/databases/main/search.py

```diff
@@ -55,7 +56,7 @@ def store_search_entries_txn(self, txn, entries):
                     entry.event_id,
                     entry.room_id,
                     entry.key,
-                    entry.value,
+                    entry.value.replace("\u0000", " "),
```

Comment:
Probably best to do this above the if-statement with a comment about why we're doing it.

```
# ...
value = entry.value.replace("\u0000", " ")
```

Reply:
So I actually tried to do this at first and then could not figure out how to do it, since `entries` is of type `SearchEntry` which is a named tuple, and then each `entry` in `entries` is a tuple itself-I was having a very hard time figuring out how to replace a value in those immutable structures. What would be the most graceful way of achieving this?

Reply:
Right, I think I completely ignored that there can be more than one entry!

Maybe a better thing to do would be to make a function which does this cleaning so we can consolidate the logic into one place and put a nice comment on it? This would also let us use it in `_find_highlights_in_postgres` which is nice!

Something like:

```python
def _clean_value_for_search(value: str) -> str:
    """..."""
    return value.replace("\u0000", " ")
```

I'm not 100% sure what to put in the docstring, but something about why we want to remove nulls as postgres breaks on those.